app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -0,0 +1,226 @@
+            return task.getGuid();
+        }
+    };
+    
+    private DynamoDBMapper mapper;
+    
+    private SchedulePlanService schedulePlanService;
+    
+    private UserConsentDao userConsentDao;
+
+    @Resource(name = "taskDdbMapper")
+    public void setDdbMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Autowired
+    public void setSchedulePlanService(SchedulePlanService schedulePlanService) {
+        this.schedulePlanService = schedulePlanService;
+    }
+    
+    @Autowired
+    public void setUserConsentDao(UserConsentDao userConsentDao) {
+        this.userConsentDao = userConsentDao;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<Task> getTasks(User user, DateTime endsOn) {
+        // Somewhat odd approach here is to reduce creating and iterating through collections.
+        List<Task> tasks = Lists.newArrayList();
+        List<Task> tasksToSave = Lists.newArrayList();
+        Set<String> guids = Sets.newHashSet();
+        
+        queryForTasks(user.getHealthCode(), tasks, guids);
+        scheduleTasksForPlans(user, endsOn, tasks, tasksToSave, guids);
+        
+        List<FailedBatch> failures = mapper.batchSave(tasksToSave);
+        BridgeUtils.ifFailuresThrowException(failures);
+        
+        finalizeTasks(tasks);
+        return tasks;
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public List<Task> getTasksWithoutScheduling(User user) {
+        List<Task> tasks = Lists.newArrayList();
+        queryForTasks(user.getHealthCode(), tasks, null);
+        finalizeTasks(tasks);
+        return tasks;
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public void updateTasks(String healthCode, List<Task> tasks) {
+        List<Task> tasksToSave = Lists.newArrayList();
+        for (Task task : tasks) {
+            task.setHealthCode(healthCode);
+            Task dbTask = mapper.load(task);
+            if (dbTask != null && (task.getStartedOn() != null || task.getFinishedOn() != null)) {
+                if (task.getStartedOn() != null) {
+                    dbTask.setStartedOn(task.getStartedOn());
+                }
+                if (task.getFinishedOn() != null) {
+                    dbTask.setFinishedOn(task.getFinishedOn());
+                }
+                tasksToSave.add(dbTask);
+            }
+        }
+        if (!tasksToSave.isEmpty()) {
+            List<FailedBatch> failures = mapper.batchSave(tasksToSave);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public void deleteTasks(String healthCode) {
+        DynamoTask hashKey = new DynamoTask();
+        hashKey.setHealthCode(healthCode);
+
+        DynamoDBQueryExpression<DynamoTask> query = new DynamoDBQueryExpression<DynamoTask>().withHashKeyValues(hashKey);
+        
+        PaginatedQueryList<DynamoTask> queryResults = mapper.query(DynamoTask.class, query);
```

Comment:
Here again, there could be multiple pages of results and you need to page through them.